### ISSUE TITLE
[Remote-SSH Bug]: latest version v0.106.1 refuses to download VS Code Server locally

### ISSUE THREAD

**Issue by BackOnTruck** [microsoft/vscode-remote-release#8931]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [ ] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It connects successfully

### Expected Behavior

First of all, I'd like to mention that there are frequent connection problems on the server side while downloading VS Code Server, so I prefer downloading locally.
In the recent versions (I tested 0.106.1 and 0.105.1), the extension seems to make two separate downloads of different server-side components, while earlier versions (I tested 0.102.0) only required one download. 
The first download works as expected (if download times out, then start locally and transfer via scp command), but in the second download, the extension insists on downloading remotely, without timing out or trying to download locally, despite the "Local Server Download" setting set to "auto".
I have to use 0.102.0 for now.

### Steps To Reproduce

1. on the server, only allow connection from the client by disabling the Internet (so that VS Code Server cannot be downloaded remotely), and remove previously installed VS Code Server
2. on the client, set "Local Server Download" to "auto", and try to connect to the server on VS Code
3. VS Code gets stuck at "Downloading VS Code Server..."

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```
[18:18:40.932] Log Level: 2
[18:18:40.941] VS Code version: 1.82.0
[18:18:40.941] Remote-SSH version: remote-ssh@0.106.1
[18:18:40.941] win32 x64
[18:18:40.955] SSH Resolver called for "ssh-remote+7b22686f73744e616d65223a225631303078342d797a7a2d31302e3235302e38312e3736227d", attempt 1
[18:18:40.955] "remote.SSH.useLocalServer": false
[18:18:40.955] "remote.SSH.showLoginTerminal": false
[18:18:40.955] "remote.SSH.remotePlatform": (REMOVED)
[18:18:40.956] "remote.SSH.path": undefined
[18:18:40.958] "remote.SSH.configFile": undefined
[18:18:40.958] "remote.SSH.useFlock": true
[18:18:40.958] "remote.SSH.lockfilesInTmp": false
[18:18:40.958] "remote.SSH.localServerDownload": auto
[18:18:40.958] "remote.SSH.remoteServerListenOnSocket": false
[18:18:40.958] "remote.SSH.showLoginTerminal": false
[18:18:40.958] "remote.SSH.defaultExtensions": []
[18:18:40.959] "remote.SSH.loglevel": 2
[18:18:40.959] "remote.SSH.enableDynamicForwarding": true
[18:18:40.959] "remote.SSH.enableRemoteCommand": false
[18:18:40.959] "remote.SSH.serverPickPortsFromRange": {}
[18:18:40.959] "remote.SSH.serverInstallPath": {}
[18:18:40.962] SSH Resolver called for host: (REMOVED)
[18:18:40.962] Setting up SSH remote "(REMOVED)"
[18:18:40.965] Using commit id "8b617bd08fd9e3fc94d14adb8d358b56e3f72314" and quality "stable" for server
[18:18:40.967] Install and start server if needed
[18:18:40.969] Checking ssh with "C:\Program Files\Common Files\Oracle\Java\javapath\ssh.exe -V"
[18:18:40.971] Got error from ssh: spawn C:\Program Files\Common Files\Oracle\Java\javapath\ssh.exe ENOENT
[18:18:40.971] Checking ssh with "C:\Windows\system32\ssh.exe -V"
[18:18:40.973] Got error from ssh: spawn C:\Windows\system32\ssh.exe ENOENT
[18:18:40.973] Checking ssh with "C:\Windows\ssh.exe -V"
[18:18:40.974] Got error from ssh: spawn C:\Windows\ssh.exe ENOENT
[18:18:40.975] Checking ssh with "C:\Windows\System32\Wbem\ssh.exe -V"
[18:18:40.976] Got error from ssh: spawn C:\Windows\System32\Wbem\ssh.exe ENOENT
[18:18:40.976] Checking ssh with "C:\Windows\System32\WindowsPowerShell\v1.0\ssh.exe -V"
[18:18:40.977] Got error from ssh: spawn C:\Windows\System32\WindowsPowerShell\v1.0\ssh.exe ENOENT
[18:18:40.977] Checking ssh with "C:\Windows\System32\OpenSSH\ssh.exe -V"
[18:18:41.001] > OpenSSH_for_Windows_8.6p1, LibreSSL 3.4.3

[18:18:41.005] Running script with connection command: "C:\Windows\System32\OpenSSH\ssh.exe" -T -D 63686 "(REMOVED)" bash
[18:18:41.008] Terminal shell path: C:\Windows\System32\cmd.exe
[18:18:41.905] > 944682c991ea: running
> ]0;C:\Windows\System32\cmd.exe
[18:18:41.905] Got some output, clearing connection timeout
[18:18:41.910] > Found existing installation at /home/(REMOVED)/.vscode-server...
[18:18:41.919] > Running ssh connection command... /home/(REMOVED)/.vscode-server/code-8b617bd08fd9e3fc94d14adb8d358b56e3f72314 command-shell --cli-data-dir /home/(REMOVED)/.vscode-server/cli --on-port --require-token 32a0432c4d37 &> "/home/(REMOVED)/.vscode-server/.cli.8b617bd08fd9e3fc94d14adb8d358b56e3f72314.log" < /dev/null
> printenv:
>     SHELL=/bin/bash
>     LC_ADDRESS=zh_CN.UTF-8
>     LC_NAME=zh_CN.UTF-8
>     LC_MONETARY=zh_CN.UTF-8
>     PWD=/home/(REMOVED)
>     LOGNAME=(REMOVED)
>     XDG_SESSION_TYPE=tty
>     MOTD_SHOWN=pam
>     HOME=/home/(REMOVED)
>     LANG=en_US.UTF-8
>     LC_PAPER=zh_CN.UTF-8
>     VSCODE_AGENT_FOLDER=/home/(REMOVED)/.vscode-server
>     SSH_CONNECTION=10.249.241.224 63687 10.250.81.76 22
>     XDG_SESSION_CLASS=user
>     LC_IDENTIFICATION=zh_CN.UTF-8
>     USER=(REMOVED)
>     SHLVL=0
>     LC_TELEPHONE=zh_CN.UTF-8
>     LC_MEASUREMENT=zh_CN.UTF-8
>     XDG_SESSION_ID=225
>     XDG_RUNTIME_DIR=/run/user/1037
>     SSH_CLIENT=10.249.241.224 63687 22
>     LC_TIME=zh_CN.UTF-8
>     PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games
> s:/usr/local/games:/snap/bin
>     DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1037/bus
>     LC_NUMERIC=zh_CN.UTF-8
>     _=/usr/bin/printenv
> Removing old logfile at /home/(REMOVED)/.vscode-server/.cli.8b617bd08fd9e3fc94d14ad
> db8d358b56e3f72314.log
[18:18:41.938] > 
> Spawned remote CLI: 2378343
> Waiting for server log...
[18:18:41.953] > 
[18:18:41.983] > 944682c991ea: start
> SSH_AUTH_SOCK====
> DISPLAY====
> listeningOn==127.0.0.1:42281==
> osReleaseId==ubuntu==
> arch==x86_64==
> vscodeArch==x64==
> bitness==64==
> tmpDir==/run/user/1037==
> platform==linux==
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> serverStartTime==49==
> 944682c991ea: end
[18:18:41.983] Received install output: 
SSH_AUTH_SOCK====
DISPLAY====
listeningOn==127.0.0.1:42281==
osReleaseId==ubuntu==
arch==x86_64==
vscodeArch==x64==
bitness==64==
tmpDir==/run/user/1037==
platform==linux==
unpackResult====
didLocalDownload==0==
downloadTime====
installTime====
serverStartTime==49==

[18:18:41.984] Remote server is listening on port 42281
[18:18:41.984] Parsed server configuration: {"serverConfiguration":{"remoteListeningOn":{"port":42281},"osReleaseId":"ubuntu","arch":"x86_64","sshAuthSock":"","display":"","tmpDir":"/run/user/1037","platform":"linux"},"serverStartTime":49,"installUnpackCode":""}
[18:18:41.986] Starting forwarding server. local port 63689 -> socksPort 63686 -> remotePort 42281
[18:18:41.987] Forwarding server listening on port 63689
[18:18:41.987] Waiting for ssh tunnel to be ready
[18:18:41.988] [Forwarding server port 63689] Got connection 0
[18:18:41.990] Tunneled port 42281 to local port 63689
[18:18:41.990] Resolved "ssh-remote+7b22686f73744e616d65223a225631303078342d797a7a2d31302e3235302e38312e3736227d" to "port 63689"
[18:18:41.991] Initizing new exec server for ssh-remote+7b22686f73744e616d65223a225631303078342d797a7a2d31302e3235302e38312e3736227d
[18:18:41.991] Resolving exec server at port 63689
[18:18:42.009] > 
[18:18:42.010] [Forwarding server port 63689] Got connection 1
[18:18:42.159] Exec server for ssh-remote+7b22686f73744e616d65223a225631303078342d797a7a2d31302e3235302e38312e3736227d created and cached
[18:18:42.162] ------




[18:18:42.181] [server] Checking /home/(REMOVED)/.vscode-server/cli/servers/Stable-8b617bd08fd9e3fc94d14adb8d358b56e3f72314/log.txt and /home/(REMOVED)/.vscode-server/cli/servers/Stable-8b617bd08fd9e3fc94d14adb8d358b56e3f72314/pid.txt for a running server...
[18:18:42.658] [server] Installing and setting up Visual Studio Code Server...
```

</p>
</details>

**Comment by TakodaS** [comment]
I had this same problem with `Stable-8b617bd08fd9e3fc94d14adb8d358b56e3f72314 version: 1.82.0` but it suddenly started working again. Probably an issue with the download server rather than any code.

**Comment by leedaekyu** [comment]
I solved the same problem as you by changing the version.

vscode 1.82.0 --> 1.72.1
ssh 0.102.0 --> 0.96

**Comment by leedaekyu** [comment]
The version of vscode 1.82 is unstable.
The ssh version is forced to 0.106. Also, you can't connect even if you change it to a different version.
The solution is to make vscode a different version.

**Comment by pWoLiAn** [comment]
I didnt see the issue after updating vscode to 1.82. it started after i updated the remote ssh extension

**Comment by crezov** [comment]
> I solved the same problem as you by changing the version.
> 
> vscode 1.82.0 --> 1.72.1 ssh 0.102.0 --> 0.96

How can I download ssh 0.96 from VSIX?

**Comment by NewtonChutney** [comment]
> likely a duplicate of #8926 . setting useExecServer to false fixed the issue for me

@pWoLiAn yes, fixed it for me..!

**Comment by NewtonChutney** [comment]
I'd rather the VSCode team puts up a notice somewhere very accessible, whenever there's an issue in such utilities..
Cause people use these tools at work, and just some random day.. They stop working as expected.. 🙁☠️

**Comment by suvadeep89** [comment]
> I didnt see the issue after updating vscode to 1.82. it started after i updated the remote ssh extension likely a duplicate of #8926 . setting useExecServer to false fixed the issue for me

This solved my issue as well, which suddenly started cropping up in version 1.82 from yesterday.

**Comment by roblourens** [comment]
This should be fixed in 0.106.2, the setting is disabled by default in vscode stable. Let me know if not. Sorry for the trouble
